### ISSUE TITLE
Uncaught Error: Class 'OCP\Lock\ILockingProvider' not found in ..

### ISSUE THREAD

**Issue by BornToBeRoot** [nextcloud/server#5381]
Browse some sites in nextcloud. See this in error log: 

```
Uncaught Error: Class 'OCP\Lock\ILockingProvider' not found in /var/www/html/nextcloud/lib/private/Files/View.php:1156 Stack trace: #0 [internal function]: OC\Files\View->OC\Files\{closure}() #1 /var/www/html/nextcloud/3rdparty/icewind/streams/src/CallbackWrapper.php(109): call_user_func(Object(Closure)) #2 [internal function]: Icewind\Streams\CallbackWrapper->stream_close() #3 {main} thrown at /var/www/html/nextcloud/lib/private/Files/View.php#1156
```
### Steps to reproduce
I can't reproduce the error... it appears sometimes in my nextcloud.log

### Server configuration

**Operating system**: CentOS 7.3 (latest)

**Web server:** Apache (latest)

**Database:** MariaDB (latest)

**PHP version:** 7.0.19

**Nextcloud version:** 11.0.3

**Updated from an older Nextcloud/ownCloud or fresh install:** ownCloud 9.1.5 migrated to a new Server and upgrade to NC 10.0.5 --> NC 11.0.3

**Where did you install Nextcloud from:** .tar.bz2 packet from nextcloud.com

**List of activated apps:**

```
Enabled:
  - activity: 2.4.1
  - dav: 1.1.1
  - federatedfilesharing: 1.1.1
  - files: 1.6.1
  - files_sharing: 1.1.1
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - user_ldap: 1.1.2
  - workflowengine: 1.1.1
```

**Nextcloud configuration:**

```
$CONFIG = array (
  'instanceid' => 'XXX',
  'passwordsalt' => 'XXX',
  'secret' => 'XXX',
  'trusted_domains' =>
  array (
    0 => 'XXX.XXX.XXX',
  ),
  'datadirectory' => '/XXX/XXX/XXX',
  'overwrite.cli.url' => 'https://XXX.XXX.XXX/',
  'dbtype' => 'mysql',
  'version' => '11.0.3.2',
  'dbname' => 'nextcloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'nextcloud',
  'dbpassword' => 'XXX',
  'installed' => true,
  'maintenance' => false,
  'singleuser' => false,
  'loglevel' => 2,
  'proxy' => 'XXX:XXX',
  'updatechecker' => false,
  'theme' => 'XXX',
  'defaultapp' => 'files',
  'default_language' => 'de_DE',
  'allow_user_to_change_display_name' => false,
  'knowledgebaseenabled' => false,
  'skeletondirectory' => '',
  'enable_previews' => false,
  'activity_expire_days' => 365,
  'htaccess.RewriteBase' => '/',
  'ldapIgnoreNamingRules' => false,
  'ldapUserCleanupInterval' => 360,
  'ldapProviderFactory' => '\\OCA\\User_LDAP\\LDAPProviderFactory',
  'filelocking.enabled' => true,
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'redis' =>
  array (
    'host' => 'localhost',
    'port' => 6379,
  ),
  'mail_smtpmode' => 'smtp',
  'mail_domain' => 'XXX.XXX',
  'mail_smtphost' => 'XXX.XXX.XXX',
  'mail_smtpport' => 'XX',
  'mail_from_address' => 'XXX',
);
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

#### Nextcloud log (data/nextcloud.log)

```
{"reqId":"WT-@iUJJ4QSqSiEnY1uHQwAAAAw","remoteAddr":"XXX.XXX.XXX.XXX","app":"PHP","message":"Uncaught Error: Class 'OCP\\Lock\\ILockingProvider' not found in \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php:1156\nStack trace:\n#0 [internal function]: OC\\Files\\View->OC\\Files\\{closure}()\n#1 \/var\/www\/html\/nextcloud\/3rdparty\/icewind\/streams\/src\/CallbackWrapper.php(109): call_user_func(Object(Closure))\n#2 [internal function]: Icewind\\Streams\\CallbackWrapper->stream_close()\n#3 {main}\n  thrown at \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php#1156","level":3,"time":"2017-06-13T15:02:33+00:00","method":"GET","url":"\/apps\/serverinfo\/update","user":"XXX-XXX-XXX...","version":"11.0.3.2"}
{"reqId":"WT-@jfWIhV2DwdCSjvERagAAAAA","remoteAddr":"XXX.XXX.XXX.XXX","app":"PHP","message":"Uncaught Error: Class 'OCP\\Lock\\ILockingProvider' not found in \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php:1156\nStack trace:\n#0 [internal function]: OC\\Files\\View->OC\\Files\\{closure}()\n#1 \/var\/www\/html\/nextcloud\/3rdparty\/icewind\/streams\/src\/CallbackWrapper.php(109): call_user_func(Object(Closure))\n#2 [internal function]: Icewind\\Streams\\CallbackWrapper->stream_close()\n#3 {main}\n  thrown at \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php#1156","level":3,"time":"2017-06-13T15:02:38+00:00","method":"GET","url":"\/settings\/ajax\/checksetup","user":"XXX-XXX-XXX...","version":"11.0.3.2"}
```
Edit:
![image](https://user-images.githubusercontent.com/16019165/27337745-a137b294-55d3-11e7-85c6-289f89cc5b53.png)

**Comment by BornToBeRoot** [comment]
Is this related and does the answer from @tflidd fix the issue?
https://help.nextcloud.com/t/ocp-lock-ilockingprovider-not-found/1728/9

**Comment by MorrisJobke** [comment]
> Is this related and does the answer from @tflidd fix the issue?
> https://help.nextcloud.com/t/ocp-lock-ilockingprovider-not-found/1728/9

Correct. This is the way to tackle this problem.

**Comment by BornToBeRoot** [comment]
After some more research, i dont think the open_basedir restrictions is enabled at all.

Doing a `grep -R open_basedir /etc/` will only show me the php.ini file where the open_basedir is commented out.

@MorrisJobke Any ideas on this? 

In the logfile i can download from NC are a few more informations... Sometimes it happens when someone download shared files, sometimes when the getstoragestats.php is called. 
If a shared file is affected by this error, i can see in the logfile, that the next to calls from out nagios also throw the exception on "/ocs/v2.php/apps/serverinfo/api/v1/info".

`
{"reqId":"xxx","remoteAddr":"xx.xx.xx.xx,"app":"PHP","message":"Uncaught Error: Class 'OCP\\Lock\\ILockingProvider' not found in \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php:1156\nStack trace:\n#0 [internal function]: OC\\Files\\View->OC\\Files\\{closure}()\n#1 \/var\/www\/html\/nextcloud\/3rdparty\/icewind\/streams\/src\/CallbackWrapper.php(109): call_user_func(Object(Closure))\n#2 [internal function]: Icewind\\Streams\\CallbackWrapper->stream_close()\n#3 {main}\n  thrown at \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php#1156","level":3,"time":"2017-07-26T10:09:57+00:00","method":"GET","url":"\/s\/SHAREID\/download?path=%2FORDNER&files=10.%20DATEI.pdf","user":"--","version":"11.0.3.2"}
{"reqId":"xxx","remoteAddr":"xx.xx.xx.xx","app":"PHP","message":"Uncaught Error: Class 'OCP\\Lock\\ILockingProvider' not found in \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php:1156\nStack trace:\n#0 [internal function]: OC\\Files\\View->OC\\Files\\{closure}()\n#1 \/var\/www\/html\/nextcloud\/3rdparty\/icewind\/streams\/src\/CallbackWrapper.php(109): call_user_func(Object(Closure))\n#2 [internal function]: Icewind\\Streams\\CallbackWrapper->stream_close()\n#3 {main}\n  thrown at \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php#1156","level":3,"time":"2017-07-26T10:15:17+00:00","method":"GET","url":"\/ocs\/v2.php\/apps\/serverinfo\/api\/v1\/info","user":"NAGIOSUSER","version":"11.0.3.2"}
{"reqId":"xxx","remoteAddr":"xx.xx.xx.xx","app":"PHP","message":"Uncaught Error: Class 'OCP\\Lock\\ILockingProvider' not found in \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php:1156\nStack trace:\n#0 [internal function]: OC\\Files\\View->OC\\Files\\{closure}()\n#1 \/var\/www\/html\/nextcloud\/3rdparty\/icewind\/streams\/src\/CallbackWrapper.php(109): call_user_func(Object(Closure))\n#2 [internal function]: Icewind\\Streams\\CallbackWrapper->stream_close()\n#3 {main}\n  thrown at \/var\/www\/html\/nextcloud\/lib\/private\/Files\/View.php#1156","level":3,"time":"2017-07-26T10:21:46+00:00","method":"GET","url":"\/ocs\/v2.php\/apps\/serverinfo\/api\/v1\/info","user":"NAGIOSUSER","version":"11.0.3.2"}
`

**Comment by MorrisJobke** [comment]
Other ideas @nickvergessen @icewind1991 ?

**Comment by nickvergessen** [comment]
No idea

**Comment by BornToBeRoot** [comment]
Seems to be fixed in version 11.0.4

Two weeks without an error...

Close?
